backpack/extensions/secondorder/diag_hessian/convnd.py

```diff
@@ -27,6 +29,41 @@ def weight(self, ext, module, g_inp, g_out, backproped):
         h_diag = torch.zeros_like(module.weight)
 
         for h_sqrt, sign in zip(sqrt_h_outs, sqrt_h_outs_signs):
-            h_diag_curr = convUtils.extract_weight_diagonal(module, X, h_sqrt, self.N)
+            h_diag_curr = convUtils.extract_weight_diagonal(
+                module, X, h_sqrt, self.N, sum_batch=True
+            )
+            h_diag.add_(sign * h_diag_curr)
+        return h_diag
+
+
+class BatchDiagHConvND(DiagHBaseModule):
+    def __init__(self, derivatives, N, params=None):
+        super().__init__(derivatives=derivatives, params=["bias", "weight"])
+        self.N = N
+
+    def bias(self, ext, module, g_inp, g_out, backproped):
+        N = module.input0.shape[0]
+        sqrt_h_outs = backproped["matrices"]
+        sqrt_h_outs_signs = backproped["signs"]
+        h_diag = torch.zeros(N, *module.bias.shape)
+
+        for h_sqrt, sign in zip(sqrt_h_outs, sqrt_h_outs_signs):
+            h_diag_curr = convUtils.extract_bias_diagonal(
+                module, h_sqrt, self.N, sum_batch=False
+            )
+            h_diag.add_(sign * h_diag_curr)
+        return h_diag
+
+    def weight(self, ext, module, g_inp, g_out, backproped):
+        N = module.input0.shape[0]
+        sqrt_h_outs = backproped["matrices"]
+        sqrt_h_outs_signs = backproped["signs"]
+        X = convUtils.unfold_by_conv(module.input0, module)
+        h_diag = torch.zeros(N, *module.weight.shape)
```

Comment:
missing device, won't work on GPU

Reply:
same as above

Reply:
will double-check, thanks for the report :)

Reply:
see above, the problem only exists for `BatchDiagH`